airbyte-integrations/connectors/destination-azure-blob-storage/src/main/java/io/airbyte/integrations/destination/azure_blob_storage/AzureBlobStorageConnectionChecker.java

```diff
@@ -0,0 +1,140 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.azure_blob_storage;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.specialized.AppendBlobClient;
+import com.azure.storage.blob.specialized.SpecializedBlobClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AzureBlobStorageConnectionChecker {
+
+  private static final String TEST_BLOB_NAME_PREFIX = "testConnectionBlob";
+  private BlobContainerClient containerClient; // schema in SQL DBs controller
+  private final AppendBlobClient appendBlobClient; // aka "SQL Table" controller
+  private final boolean overwriteDataInStream;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(
+      AzureBlobStorageConnectionChecker.class);
+
+  public AzureBlobStorageConnectionChecker(
+                                           AzureBlobStorageDestinationConfig azureBlobStorageConfig,
+                                           boolean overwriteDataInStream) {
+
+    this.overwriteDataInStream = overwriteDataInStream;
+
+    StorageSharedKeyCredential credential = new StorageSharedKeyCredential(
+        azureBlobStorageConfig.getAccountName(),
+        azureBlobStorageConfig.getAccountKey());
+
+    this.appendBlobClient =
+        new SpecializedBlobClientBuilder()
+            .endpoint(azureBlobStorageConfig.getEndpointUrl())
+            .credential(credential)
+            .containerName(azureBlobStorageConfig.getContainerName()) // Like schema in DB
+            .blobName(TEST_BLOB_NAME_PREFIX + UUID.randomUUID()) // Like table in DB
+            .buildAppendBlobClient();
+  }
+
+  // this a kinda test method that is used in CHECK operation to make sure all works fine with the
+  // current config
+  public void attemptWriteAndDelete() {
+    initTestContainerAndBlob();
+    writeUsingAppendBlock("Some test data");
+    listBlobsInContainer()
+        .forEach(
+            blobItem -> LOGGER.debug(
```

Comment:
If this method is used during CHECK operation, maybe it it better to use `LOGGER.info()`
What do you think?

Reply:
And same for other `debug` levels in this class

Reply:
done, thanks!